libs/ui/src/invalid_card_screen.test.tsx

```diff
@@ -1,11 +1,71 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 
-import { InvalidCardScreen } from './invalid_card_screen';
+import { InvalidCardScreen, Props } from './invalid_card_screen';
 
-describe('InvalidCardScreen', () => {
-  test('says "Invalid Card"', () => {
-    const { getByText } = render(<InvalidCardScreen />);
-    getByText('Invalid Card');
-  });
-});
+const testCases: Array<{
+  description: string;
+  machine: Props['machine'];
+  reason: Props['reason'];
+  expectedText: string;
+}> = [
+  {
+    description: 'VxAdmin not configured',
+    machine: 'VxAdmin',
+    reason: 'machine_not_configured',
+    expectedText:
+      'This machine is unconfigured and cannot be unlocked with this card. ' +
+      'Please insert a System Administrator card.',
+  },
+  {
+    description: 'VxCentralScan not configured',
+    machine: 'VxCentralScan',
+    reason: 'machine_not_configured',
+    expectedText:
+      'This machine is unconfigured and cannot be unlocked with this card. ' +
+      'Please insert an Election Manager card.',
+  },
+  {
+    description:
+      'Election Manager card election hash does not match VxAdmin election hash',
+    machine: 'VxAdmin',
+    reason: 'admin_wrong_election',
+    expectedText:
+      'The inserted Election Manager card is programmed for another election and cannot be used to unlock this machine. ' +
+      'Please insert a valid Election Manager or System Administrator card.',
+  },
+  {
+    description:
+      'Election Manager card election hash does not match VxCentralScan election hash',
+    machine: 'VxCentralScan',
+    reason: 'admin_wrong_election',
+    expectedText:
+      'The inserted Election Manager card is programmed for another election and cannot be used to unlock this machine. ' +
+      'Please insert a valid Election Manager or System Administrator card.',
+  },
+  {
+    description: 'other error on VxAdmin',
+    machine: 'VxAdmin',
+    reason: 'invalid_user_on_card',
+    expectedText:
+      'The inserted card is not valid to unlock this machine. ' +
+      'Please insert a valid Election Manager or System Administrator card.',
+  },
+  {
+    description: 'other error on VxCentralScan',
+    machine: 'VxCentralScan',
+    reason: 'invalid_user_on_card',
+    expectedText:
+      'The inserted card is not valid to unlock this machine. ' +
+      'Please insert a valid Election Manager or System Administrator card.',
+  },
+];
+
+test.each(testCases)(
```

Comment:
Decided to experiment with jest's built-in tooling for table-driven tests - it worked well! The test output for reference:

```
 PASS  src/invalid_card_screen.test.tsx
  ✓ InvalidCardScreen renders expected text (VxAdmin not configured) (27 ms)
  ✓ InvalidCardScreen renders expected text (VxCentralScan not configured) (3 ms)
  ✓ InvalidCardScreen renders expected text (Election Manager card election hash does not match VxAdmin election hash) (2 ms)
  ✓ InvalidCardScreen renders expected text (Election Manager card election hash does not match VxCentralScan election hash) (3 ms)
  ✓ InvalidCardScreen renders expected text (other error on VxAdmin) (2 ms)
  ✓ InvalidCardScreen renders expected text (other error on VxCentralScan) (2 ms)

Test Suites: 1 passed, 1 total
Tests:       6 passed, 6 total
Snapshots:   0 total
Time:        2.784 s
```